pheps/phep-0001.md

```diff
@@ -0,0 +1,287 @@
+8. How to Teach This - For a PHEP that adds new functionality or changes behavior, it is helpful to include a section on how to teach users, new and experienced, how to apply the PHEP to their work.
+
+    This section may include key points and recommended documentation changes that would help users adopt a new feature or migrate their code to use a language change.
+
+9. Reference Implementation - The reference implementation must be completed before any PHEP is given status "Final", but it need not be completed before the PHEP is accepted. While there is merit to the approach of reaching consensus on the specification and rationale before writing code, the principle of "rough consensus and running code" is still useful when it comes to resolving many discussions of API details.
+
+    The final implementation must include test code and documentation appropriate for the relevant PyHC project(s).
+
+10. Rejected Ideas - Throughout the discussion of a PHEP, various ideas will be proposed which are not accepted. Those rejected ideas should be recorded along with the reasoning as to why they were rejected. This both helps record the thought process behind the final version of the PHEP as well as preventing people from bringing up the same rejected idea again in subsequent discussions.
+
+    In a way this section can be thought of as a breakout section of the Rationale section that is focused specifically on why certain ideas were not ultimately pursued.
+
+11. Open Issues - While a PHEP is in draft, ideas can come up which warrant further discussion. Those ideas should be recorded so people know that they are being thought about but do not have a concrete resolution. This helps make sure all issues required for the PHEP to be ready for consideration are complete and reduces people duplicating prior discussion.
+12. Footnotes - A collection of footnotes cited in the PHEP, and a place to list non-inline hyperlink targets.
+13. Copyright/license - Each new PHEP must be placed under a dual license of public domain and [CC0-1.0-Universal](https://choosealicense.com/licenses/cc0-1.0/) (see this PHEP for an example).
+
+# PHEP Formats and Templates
+PHEPs are UTF-8 encoded text files using the Markdown format. Markdown allows for rich markup that is still quite easy to read, but also results in good-looking and functional HTML. A future PHEP may include a template. PHEP Markdown must be compatible with the most recent version of the [CommonMark specification](https://spec.commonmark.org/) at the time the PHEP is approved. PHEPs must not use [GitHub extensions](https://github.github.com/gfm/).
+
+PHEPs should be restricted to the ASCII subset of UTF-8 where practicable. Examples of cases where it is not practicable include personal names.
+
+The PHEP text files are automatically rendered by the GitHub interface for easier online reading. The Markdown source is the version of record.
+
+The readability of the diff should be considered when editing a PHEP.
```

Comment:
Really need input on this. As written each paragraph is one megaline. xxdiff gives something usable from this, but e.g. gitk doesn't. If we line break in the middle of the paragraph we need rules for reflow--my thought would be something like "start with 72 character lines, don't reflow when editing, and have one commit that's nothing but reflow" but that is really finicky.

Reply:
Yeah, that's finicky and I don't see people liking those rules too much. I personally think it's sufficient to just say "the readability of the diff should be considered when editing a PHEP." For me, having to think too hard in advance about how GitHub will render my diff would be more cumbersome than just dealing with not-super-great diffs.

Reply:
I thought markdown could handle one sentence per line and that would render as one paragraph? You shouldn't need a line break.

Reply:
Markdown will render with word wrap just fine. The difficulty is when the markdown source isn't wrapped (as is the case for this PR, guilty as charged) the diff gets to be ridiculous because the whole "line" is a changed line if a single character changes. xxdiff and vimdiff try to render sub-line diffs with varying degrees of success, as does GitHub (see the links in the main comment), but standard `git diff` and gitk don't.

Doing one sentence per line sounds pretty reasonable--it'll reflow fine when rendered and lumps "changed" lines with actually related content. It would be a bit weird/ragged when reading the source directly but probably not too bad.

Reply:
Fwiw I don't think we need to prioritize the source directly looking nice, so long as the document renders well.

Reply:
This is less about the source looking nice and more about being able to follow the set of changes through the diffs as the text evolves (although I know you're responding to my note).

While I have written a lot of poorly-understood sentences in my life, and this is probably not even in the top ten, it certainly doesn't seem to be _helping_. So I'm going to axe this. It's up to the author to manage the discussion; they can certainly take some of these ideas into consideration. And the editor can help if things seem to be going down a difficult-to-follow path.

Similarly, I am going to axe the notion that the markdown source is the version of record. It's persnickety and if we run into trouble of that sort, we'll have to sort it out then.

Reply:
Sounds great.

("I have written a lot of poorly-understood sentences in my life"... lol)